### PR TITLE
use input layers title as default impact layer title

### LINKED ISSUE #497: auto-generated impact layer name
# problem

If a user does not specify a name for their impact layer, the system concocts one from the layer name fields (?) of the input layers, which looks terrible. 

# proposed solution

construct the impact layer name from the title fields so it's more human readable. Maybe include mention of the aggregation title if one was used

### PR DESCRIPTION
**Pull request by myarjunar**
fix #497

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -307,9 +307,12 @@ def get_task_state(self):
 
     def get_default_impact_title(self):
         layer_name = '%s on %s' % (
```
Comment (lucernae) [inline models.py]: Title needs to be translated as well if possible. So, use format string with context on it, such as:
```suggestion
        layer_name = '{hazard} on {exposure}'.format(
            hazard=self.hazard_layer.title,
            exposure=self.exposure_layer.title)
```

File: models.py [hunk 2]
```diff
@@ -307,9 +307,12 @@ def get_task_state(self):
...
-            self.hazard_layer.name,
-            self.exposure_layer.name
+            self.hazard_layer.title,
+            self.exposure_layer.title
         )
+        if self.aggregation_layer:
+            additional_name = 'around %s' % self.aggregation_layer.title
```
Comment (lucernae) [inline models.py]: Same reason as above, use:

```suggestion
            additional_name = 'around {aggregation}'.format(aggregation=self.aggregation_layer.title)
```

File: models.py [hunk 3]
```diff
@@ -307,9 +307,12 @@ def get_task_state(self):
...
+            return layer_name + ' ' + additional_name
```
Comment (lucernae) [inline models.py]: In fact, considering this end result, it is better to do it like this (for this whole method):

```suggestion
            title_dict = {
                'hazard': self.hazard_layer.title,
                'exposure': self.exposure_layer.title
            }
            if self.aggregation_layer:
                title_template = _('{hazard} on {exposure} around {aggregation}')
                title_dict['aggregation'] = self.aggregation_layer.title
            else:
                title_template = _('{hazard} on {exposure}')
             return title_template.format(**title_dict)
                
```

Comment (lucernae) [reply models.py]: This way, the translator has context and shuffle the variable around if needed.

**Review by lucernae — APPROVED** [review]
LGTM, thanks @myarjunar but please read my comment here.

**Comment by myarjunar** [comment]
I will translate new strings in this PR on another PR. I will merge this first.

**Pull request merged**